src/main/java/seedu/address/model/entry/Link.java

```diff
@@ -53,14 +45,14 @@ public static String formExceptionMessage(String invalidInput) {
 
     @Override
     public String toString() {
-        return value;
+        return value.toString();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Link // instanceof handles nulls
-                && value.equals(((Link) other).value)); // state check
+                && value.toString().toLowerCase().equals(((Link) other).value.toString().toLowerCase())); // state check
```

Comment:
We check equality of URLs after down-casing, since this is how we hash URLs in ArticleStorage